mailbox/api/src/main/java/org/apache/james/mailbox/model/SearchQuery.java

```diff
@@ -594,10 +595,18 @@ public static Criterion flagSet(Flag flag, boolean isSet) {
         return result;
     }
 
+    public static Criterion hasAttachment() {
+        return new AttachmentCriterion(BooleanOperator.set());
+    }
+
+    public static Criterion hasNoAttachment() {
```

Comment:
The work of handling both is almost free in all implementation. I proposed two method hasAttachment and hasNoAttachment mimicing hasFlag and hasNoFlag, to have a consistent API. As a potential futur user of this API, I think it's nice to have a method directly for the negation